community/values/src/main/java/org/neo4j/values/storable/PointValue.java

```diff
@@ -361,28 +361,37 @@ public static PointValue parse( CharSequence text )
         return fromInputFields( fields );
     }
 
-    /**
-     * This contains the logic to decide the default coordinate reference system based on the input fields
-     */
-    private static PointValue fromInputFields( AnyValue[] fields )
+    private static CoordinateReferenceSystem findSpecifiedCRS( AnyValue[] fields )
     {
-        CoordinateReferenceSystem crs;
-        double[] coordinates;
-
         AnyValue crsValue = fields[PointValueField.CRS.ordinal()];
-        if ( crsValue != null )
+        AnyValue sridValue = fields[PointValueField.SRID.ordinal()];
+        if ( crsValue != null && sridValue != null )
+        {
+            throw new IllegalArgumentException( "Cannot specify both CRS and SRID" );
+        }
+        else if ( crsValue != null )
         {
             TextValue crsName = (TextValue) crsValue;
-            crs = CoordinateReferenceSystem.byName( crsName.stringValue() );
-            if ( crs == null )
-            {
-                throw new IllegalArgumentException( "Unknown coordinate reference system: " + crsName.stringValue() );
-            }
+            return CoordinateReferenceSystem.byName( crsName.stringValue() );
```

Comment:
Why did you remove the null-check with error message `"Unknown coordinate reference system: " + crsName.stringValue()` for this case?

Reply:
Because I noticed that the CoordinateReferenceSystem.* methods never return null. They all throw the same exception we were throwing here, with the same error message even. So we actually had two checks in a row. I've removed the redundancy.